Makefile

```diff
@@ -1,17 +1,44 @@
-BINARY_NAME=fga
-BUILD_FOLDER=dist
+#-----------------------------------------------------------------------------------------------------------------------
+# Variables (https://www.gnu.org/software/make/manual/html_node/Using-Variables.html#Using-Variables)
+#-----------------------------------------------------------------------------------------------------------------------
+BINARY_NAME = fga
+BUILD_DIR ?= $(CURDIR)/dist
```

Comment:
Using `$(CURDIR)` in Makefiles is preferred over using the shell command `pwd` or the relative path `./ `for indicating the current directory due to a few key reasons related to consistency, reliability, and make's built-in features:

- **Consistency Across Platforms**: `$(CURDIR)` is a built-in variable in GNU make that reliably points to the directory where the Makefile is located, regardless of the platform or shell environment. This consistency is crucial for writing portable Makefiles that work across different systems and setups.

- **Reliability in Sub-Make Calls**: When using recursive make calls (calling make from a Makefile), `$(CURDIR)` ensures that you are always referring to the directory of the top-level Makefile, not the directory where make was invoked. This is particularly important in complex projects with nested Makefiles, where the relative path` ./ `or the output of `pwd` might not point to the directory you expect.

- **Avoiding Shell Dependency**: Using `pwd` requires a shell invocation, which can be less efficient and potentially inconsistent depending on the user's shell configuration. In contrast, `$(CURDIR)` is a make built-in variable, so it doesn't spawn a new shell process, leading to faster and more predictable execution.